minari/utils.py

```diff
@@ -1,20 +1,204 @@
+        else:
+            if inclusion_interval.left == P.CLOSED:
+                operator = ">="
+            else:
+                operator = ">"
+            final_version_specifier &= f"{operator}{str(inclusion_interval.lower)}"
+
+            if inclusion_interval.right == P.CLOSED:
+                operator = "<="
+            else:
+                operator = "<"
+            final_version_specifier &= f"{operator}{str(inclusion_interval.upper)}"
+    else:
+        if inclusion_interval.lower != -P.inf:
+            if inclusion_interval.left == P.CLOSED:
+                operator = ">="
+            else:
+                operator = ">"
+            final_version_specifier &= f"{operator}{str(inclusion_interval.lower)}"
+        if inclusion_interval.upper != P.inf:
+            if inclusion_interval.right == P.CLOSED:
+                operator = "<="
+            else:
+                operator = "<"
+            final_version_specifier &= f"{operator}{str(inclusion_interval.upper)}"
+
+    # If the versions to be excluded fall inside the previous calculated version specifier
+    # add them to the specifier as `!=`
+    for spec in exclusion_specifiers:
+        version = spec.version
+        if version[-1] == "*":
+            version = Version(version[:-2])
+            release = list(version.release)
+            release[-1] = 0
+            release[-2] += 1
+            max_release = Version(".".join(str(r) for r in release))
+            exclusion_interval = P.closedopen(version, max_release)
+            if inclusion_interval.overlaps(exclusion_interval):
+                final_version_specifier &= str(spec)
+        elif version in final_version_specifier:
+            final_version_specifier &= str(spec)
+
+    return final_version_specifier
+
+
+def validate_datasets_to_combine(datasets_to_combine: List[MinariDataset]):
+    """Check if the given datasets can be combined.
+
+    Tests if the datasets were created with the same environment (`env_spec`) and re-calculates the
+    `max_episode_steps` argument.
+
+    Args:
+        datasets_to_combine (List[MinariDataset]): list of MinariDataset to combine
+
+    Returns:
+        combined_dataset_env_spec (EnvSpec): the resulting EnvSpec of combining the MinariDatasets
+    """
+    combined_dataset_env_spec = None
+
+    for dataset in datasets_to_combine:
+        if not isinstance(dataset, MinariDataset):
+            raise ValueError(f"The dataset {dataset} is not of type MinariDataset.")
+        dataset_env_spec = dataset.spec.env_spec
+
+        assert isinstance(dataset_env_spec, EnvSpec)
+
+        # Look for the maximum value of `max_episode_steps` of the environments of all the
+        # datasets to combine and sets that value to the `max_episode_steps` of the combined dataset.
+        if combined_dataset_env_spec is None:
+            combined_dataset_env_spec = dataset_env_spec
+        elif dataset_env_spec.max_episode_steps is not None:
```

Comment:
If some of `dataset_env_spec.max_episode_steps` are None and some no, you ignore the ones with `None`. Is this the intended behavior? Or should we propagate the None?

And, why can a dataset max_episode_steps be None?

Reply:
Agree. `None` should be an indefinite number of steps so it should propagate. I'll fix this.